model/github_app_auth.go

```diff
@@ -0,0 +1,133 @@
+		"op":         opName,
+		"project_id": appAuth.Id,
+		"epic":       "DEVPROD-5552",
+	}))
+	if !isPSEnabled {
+		return
+	}
+
+	op(ref, isRepoRef)
+}
+
+// GitHubAppAuthUpsert upserts the GitHub app auth into the database and to
+// Parameter Store if enabled.
+func GitHubAppAuthUpsert(appAuth *githubapp.GithubAppAuth) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultParameterStoreAccessTimeout)
+	defer cancel()
+
+	githubAppCheckAndRunParameterStoreOp(ctx, appAuth, func(ref *ProjectRef, isRepoRef bool) {
+		paramName, err := githubAppAuthUpsertParameterStore(ctx, appAuth, ref, isRepoRef)
+		grip.Error(message.WrapError(err, message.Fields{
+			"message":    "could not upsert GitHub app auth into Parameter Store",
+			"op":         "Upsert",
+			"project_id": appAuth.Id,
+			"epic":       "DEVPROD-5552",
+		}))
+		if paramName != "" {
+			appAuth.PrivateKeyParameter = paramName
+		}
+	}, "Upsert")
+
+	return githubapp.UpsertGithubAppAuth(appAuth)
+}
+
+// githubAppAuthUpsertParameterStore upserts the GitHub app auth into Parameter
+// Store.
+func githubAppAuthUpsertParameterStore(ctx context.Context, appAuth *githubapp.GithubAppAuth, pRef *ProjectRef, isRepoRef bool) (string, error) {
+	projectID := appAuth.Id
+	partialParamName := getPrivateKeyParamName(projectID)
+
+	paramMgr := evergreen.GetEnvironment().ParameterManager()
+	paramValue := string(appAuth.PrivateKey)
+
+	param, err := paramMgr.Put(ctx, partialParamName, paramValue)
+	if err != nil {
+		return "", errors.Wrap(err, "upserting GitHub app private key into Parameter Store")
+	}
+	paramName := param.Name
+
+	existingParamName := appAuth.PrivateKeyParameter
+	if existingParamName != "" && existingParamName != paramName {
+		if err := paramMgr.Delete(ctx, existingParamName); err != nil {
+			return "", errors.Wrapf(err, "deleting old GitHub app private key parameter '%s' from Parameter Store after it was renamed to '%s'", existingParamName, paramName)
+		}
+	}
+
+	if err := pRef.setParameterStoreGitHubAppAuthSynced(true, isRepoRef); err != nil {
+		return "", errors.Wrapf(err, "marking project/repo ref '%s' as having its GitHub app auth synced to Parameter Store", pRef.Id)
+	}
+
+	return paramName, nil
+}
+
+// GitHubAppAuthRemove removes the GitHub app auth from the database and from
+// Parameter Store if enabled.
+func GitHubAppAuthRemove(appAuth *githubapp.GithubAppAuth) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultParameterStoreAccessTimeout)
+	defer cancel()
+
+	githubAppCheckAndRunParameterStoreOp(ctx, appAuth, func(ref *ProjectRef, isRepoRef bool) {
+		if err := githubAppAuthRemoveParameterStore(ctx, appAuth); err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":    "could not delete GitHub app auth from Parameter Store",
+				"op":         "Remove",
+				"project_id": appAuth.Id,
+				"epic":       "DEVPROD-5552",
+			}))
+		}
+	}, "Remove")
+	return githubapp.RemoveGithubAppAuth(appAuth.Id)
```

Comment:
Should we skip this DB update if we fail to delete the resource in Parameter store? Same q for upsert

Reply:
If we fail to delete it, I think it's okay for now to just continue. If there's a problem with Parameter Store, I wouldn't want that to block the removal of their app auth. The operation will eventually return if it fails to remove the parameter, once we're more confident that the Parameter Store integration is working.